File: python/formats/tileset.py
# coding=UTF-8
from binaryio import readUnsignedByte, readInt

tilePath = './'

# ----
# Read a tileset file.
# - 
# Notes:	Only handles a detail level of 1 (RGB, 32x32 tiles)
# ----
def load(tstFileName):
	tstFile = open(tilePath + tstFileName, 'rb')
	
	# Read the tileset header:
	thisTileset = dict()
	thisTileset['filename'] = tstFileName
	thisTileset['version'] = readInt(tstFile)
	thisTileset['count'] = readInt(tstFile)
	thisTileset['detail'] = readInt(tstFile)

	if thisTileset['detail'] != 1:
		print 'Sorry - Cannot read anything other than detail level 1 tilesets'
		return

	thisTileset['tiles'] = []

	for tile in range(thisTileset['count']):
		thisTile = []
		for x in range(32):
			thisX = []
			for y in range(32):
				r = readUnsignedByte(tstFile)
				g = readUnsignedByte(tstFile)
				b = readUnsignedByte(tstFile)

				thisX.append((r,g,b))
			thisTile.append(thisX)

		thisTileset['tiles'].append(thisTile)
	
	tstFile.close()

	return thisTileset

File: php/formats/tileset.php
<?php

class Tileset
{
	public static $tilePath = './';

	public static function load($tilesetFile)
	{
		$file = new BinaryIO(self::$tilePath . $tilesetFile);

		$thisTileset = array();
		$thisTileset['filename'] = $tilesetFile;
		$thisTileset['version'] = $file->readInt();
		$thisTileset['count'] = $file->readInt();
		$thisTileset['detail'] = $file->readInt();

		if($thisTileset['detail'] != 1)
		{
			throw new Exception('Cannot read tilesets with a detail level other than 1.');
		}

		$thisTileset['tiles'] = array();

		for($i = 0; $i < $thisTileset['count']; $i++)
		{
			$thisTile = array();

			for($x = 0; $x < 32; $x++)
			{
				$thisX = array();

				for($y = 0; $y < 32; $y++)
				{
					$r = $file->readUnsignedByte();
					$g = $file->readUnsignedByte();
					$b = $file->readUnsignedByte();

					$thisX[] = array($r, $g, $b);
					unset($r, $g, $b);
				}

				$thisTile[] = $thisX;
				unset($thisX);
			}

			$thisTileset['tiles'][] = $thisTile;
			unset($thisTile);
		}

		return $thisTileset;
	}
}